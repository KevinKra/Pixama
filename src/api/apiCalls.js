import API_KEY from "./apikey";

export const fetchPopularMovies = async () => {
  let response = await fetch(
    `https://api.themoviedb.org/3/discover/movie?certification_country=US&api_key=${API_KEY}`
  );
  const movies = await response.json();
  return movies.results;
};

//This method is not hooked up anywhere at this point
//data must be an object with keys of name, email and password
export const fetchNewUser = (url, data)  => {
  return fetch(url, 
    {
      method: 'POST',
      body: JSON.stringify( data ),
      headers: {
        'Content-Type': 'application/json'
    }
  })
  .then(response=> {
    if(!response.ok) {
      throw new Error ("Email already exists", response.message)
    } else {
      return response.json()
    }
  })
}

//this method is being imported and called in LoginCard component
export const fetchUser = (url, data)  => {

  return fetch(url, 
    {
      method: 'POST',
      body: JSON.stringify( data ),
      headers: {
        'Content-Type': 'application/json'
    }
  })
  .then(response=> {
    if(!response.ok) {
      throw new Error ("Not a valid login",response.message)
    } else {
      return response.json()
    }
  })
}

//Method to add a movie to a user's favorites, this method not hooked up to a component yet
// url in component method needs to be 'http://localhost:3000/api/users/favorites/new'
//data needs to be object like this: {movie_id: 550, user_id: 1, title: "Fight Club", poster_path: "/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg" , release_date: "1999-10-15", vote_average:8.4, overview: "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground \"fight clubs\" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion."}
//response will be the new favorite ID
export const postFavorite = (url, data)  => {
console.log(data)
return fetch(url, 
    {
      method: 'POST',
      body: JSON.stringify( data ),
      headers: {
        'Content-Type': 'application/json'
    }
  })
  .then(response=> {
    if(!response.ok) {
      throw new Error ("Add error",response.message)
    } else {
      return response.json()
    }
  })
};


export const fetchFavorites = (url)  => {

return fetch(url)
  .then(response=> {
    if(!response.ok) {
      throw Error ("Couldn't get favorites",response.message)
    } else {
      return response.json()
    }
  })
};