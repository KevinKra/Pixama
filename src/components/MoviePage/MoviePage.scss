.MoviePage {
  position: relative;
  min-height: 100vh;
  min-width: 100vw;
  display: flex;
  flex-direction: column;
}

.mp-primary-content {
  position: relative;
  display: flex;
  margin: 0 auto;
  width: 100%;
  color: white;
  aside {
    position: relative;
    animation: slideInRight 1s forwards;
    width: 500px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
  .poster-image {
    position: relative;
    z-index: 2;
    animation: slideInLeft 1s forwards;
    border: 1px solid black;
    box-shadow: 5px 5px 10px black;
    height: 100vh;
    width: 35vw;
    cursor: pointer;
  }
  .backdrop-image {
    position: relative;
    z-index: 2;
    animation: fadeIn 1s forwards;
    border: 1px solid black;
    box-shadow: 5px 5px 10px black;
    height: 100vh;
    width: 60vw;
    cursor: pointer;
  }
  .production-companies {
    justify-self: flex-end;
    text-align: center;
  }
  .production-companies > * {
    cursor: pointer;
    margin: 1rem;
    vertical-align: middle;
  }
  img {
    filter: invert(100%);
  }
}

.backdrop {
  border: 1px solid red;
  height: 100vh;
  width: 100vw;
  position: absolute;
  background: url("https://images.unsplash.com/photo-1482160549825-59d1b23cb208?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80");
  opacity: 0.02;
  filter: grayscale(1) invert(100%);
}

.mp-support-content {
  border: 1px solid blue;
}

.mp-movie-info {
  // border: 1px solid red;
  padding-left: 1rem;
  .movie-support {
    display: flex;
    justify-content: center;
  }
  .movie-support > * {
    margin: 0 1rem;
  }
}

@keyframes slideInRight {
  from {
    transform: translateX(250%);
  }
  to {
    transform: translateX(0%);
  }
}

@keyframes slideInLeft {
  from {
    transform: translateX(-250%);
  }
  to {
    transform: translateX(0%);
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
