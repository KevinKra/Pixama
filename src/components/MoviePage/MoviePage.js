import React, { Component } from "react";
import * as apiCalls from "../../api/apiCalls";
import API_KEY from "../../api/apikey";
import "./MoviePage.scss";

export default class MoviePage extends Component {
  state = {
    movie: {},
    displayPoster: true
  };

  fetchFightClub = async () => {
    let response = await fetch(
      `https://api.themoviedb.org/3/movie/550?api_key=${API_KEY}`
    );
    const movies = await response.json();
    return movies;
  };

  async componentDidMount() {
    const movie = await this.fetchFightClub();
    this.setState({ movie });
  }

  togglePoster = () => {
    console.log("clicky");
    const toggle = this.state.displayPoster;
    this.setState({ displayPoster: !toggle });
  };

  divStyle = (image, position) => {
    return {
      backgroundImage: `url(${image})`,
      backgroundPosition: position,
      backgroundSize: "cover",
      backgroundRepeat: "no-repeat"
    };
  };

  render() {
    const {
      poster_path,
      backdrop_path,
      overview,
      original_title,
      original_language,
      release_date,
      runtime,
      status,
      revenue,
      homepage,
      tagline,
      vote_average,
      production_companies,
      id
    } = this.state.movie;
    const posterToggle = () => {
      return this.state.displayPoster ? (
        <div
          className="poster-image"
          onClick={this.togglePoster}
          style={this.divStyle(
            `https://image.tmdb.org/t/p/original${poster_path}`,
            "center top"
          )}
          key={id}
          id={id}
        />
      ) : (
        <div
          className="backdrop-image"
          style={this.divStyle(
            `https://image.tmdb.org/t/p/original${backdrop_path}`,
            "center top"
          )}
          onClick={this.togglePoster}
          key={id}
          id={id}
        />
      );
    };
    return (
      <section className="MoviePage">
        <section className="mp-primary-content">
          {posterToggle()}
          <div className="backdrop" />
          <aside>
            <section className="mp-movie-info">
              <h1>{original_title}</h1>
              <h3>{tagline}</h3>
              <div className="movie-support">
                <p>favorited</p>
                <p>Rating: {vote_average}</p>
                <p>Length: {runtime}</p>
              </div>
              <p>{overview}</p>
              <p>{status}</p>
              <p>Released: {release_date}</p>
              <p>Revenue: {revenue}</p>
              <p>Language: {original_language}</p>
              <a href={homepage} />
            </section>
            <section className="production-companies">
              {production_companies &&
                production_companies.map(
                  (company, i) =>
                    company.logo_path && (
                      <img
                        src={`https://image.tmdb.org/t/p/w92${
                          company.logo_path
                        }`}
                        alt={company.name}
                        key={i}
                      />
                    )
                )}
            </section>
          </aside>
        </section>
      </section>
    );
  }
}
