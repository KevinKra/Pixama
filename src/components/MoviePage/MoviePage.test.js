import { MoviePage } from "./MoviePage";
import React from "react";
import { shallow } from "enzyme";
import * as apiCalls from "../../api/apiCalls";


describe("MoviePage", () => {
  let wrapper;
  let instance;
  let mockMoviePage = {
    poster_path: "awregawg",
    backdrop_path: "argas",
    overview: "asrg",
    original_title: "gkudjy",
    original_language: "dfyjdf",
    release_date: "gf,sbd",
    runtime: "dfbs",
    status: "dsbsdb",
    revenue: 1343,
    tagline: "sdfbsd",
    vote_average: 314,
    production_companies: [
      { logo_path: "woiurhgwr" },
      { logo_path: "WDwd" }
    ],
    id: 123
  };

  beforeEach(() => {
    wrapper = shallow(<MoviePage moviePage={mockMoviePage} />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it ('should toggle displayPoster in state with togglePoster', () => {
    const before = wrapper.state("displayPoster");
    instance.togglePoster();
    const after = wrapper.state("displayPoster");
    expect (before).not.toEqual(after);
  });
  
});