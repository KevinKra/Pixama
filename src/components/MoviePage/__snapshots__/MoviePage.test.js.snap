// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should match the snapshot 1`] = `
<section
  className="MoviePage"
>
  <section
    className="mp-primary-content"
  >
    <div
      className="poster-image"
      id={123}
      key="123"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/originalawregawg)",
          "backgroundPosition": "center top",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="backdrop"
    />
    <aside>
      <section
        className="mp-movie-info"
      >
        <h1>
          gkudjy
        </h1>
        <h3>
          sdfbsd
        </h3>
        <div
          className="movie-support"
        >
          <p>
            favorited
          </p>
          <p>
            Rating: 
            314
          </p>
          <p>
            Length: 
            dfbs
          </p>
        </div>
        <p>
          asrg
        </p>
        <p>
          dsbsdb
        </p>
        <p>
          Released: 
          gf,sbd
        </p>
        <p>
          Revenue: 
          1343
        </p>
        <p>
          Language: 
          dfyjdf
        </p>
      </section>
      <section
        className="production-companies"
      >
        <img
          key="0"
          src="https://image.tmdb.org/t/p/w92woiurhgwr"
        />
        <img
          key="1"
          src="https://image.tmdb.org/t/p/w92WDwd"
        />
      </section>
    </aside>
  </section>
</section>
`;
